### ISSUE TITLE
Change to newer electron

### ISSUE THREAD

**Issue by Jogai** [tareqimbasher/NetPad#270]
Seems easy to upgrade to [ElectronSharp](https://github.com/theolivenbaum/electron-sharp) from ElectronNET to make use of newer dependencies. Would you be interested in that?

**Comment by SimonNyvall** [comment]
There was a PR introducing Tauri (#265), which brought a new Rust-based shell as an alternative to the Electron version. From my understanding, the project is maintaining two versions of the app for now: one using Electron and another using Tauri.

That said, I’m not sure what the long-term plans are for the Electron version or any upgrades, maybe @tareqimbasher has some insights for future plans?

**Comment by Jogai** [comment]
Thanks for the clarification. I was triggered because the arch aur distrubutes an electron version.

**Comment by SimonNyvall** [comment]
Yeah, the bin uploaded to the AUR is version `0.7.2` and only ships with the Electron shell. There is an AppImage available for [download](https://github.com/tareqimbasher/NetPad/releases/tag/v0.8.0) with the Tauri shell, but I have not tried it yet.

**Comment by tareqimbasher** [comment]
I think the plan is to continue maintaining the Electron version for a while, but the idea is to eventually phase it out in favor of the more lightweight Tauri one. There isn't a specific timeline; its really when the Tauri version is stable. I was thinking of moving to ElectronSharp due tot Electron.NET's lack of maintenance but now I'm not sure that effort is worth it, unless there is a real benefit (example: faster startup) from moving to a later Electron version.

@Jogai I don't maintain the AUR package so I'm not sure if `electron23` _should_ be a dependency to be able to run NetPad since its offering the pre-built binary, try taking it out of the `PKGBUILD` script and see if it will run without it. Or alternatively try removing just `electron23` (if nothing else depends on it besides NetPad) and see if it runs.

**Comment by Jogai** [comment]
> I'm not sure that effort is worth it, unless there is a real benefit

The effort required is really low imo, and the benefits would be at least to not include a dependency with a fair bit of [vulnerabilities](https://security.snyk.io/package/npm/electron/23.2.0).
However those are only fixed in versions supported by recent electron sharp releases which require net8. Since net6 is still supported for another month I wonder if you would be willing to move to a net8 only version. If that's the case I can make a separate PR for just updating electron?

**Comment by Jogai** [comment]
I did both, see: https://github.com/tareqimbasher/NetPad/pull/274

**Comment by tareqimbasher** [comment]
@Jogai thanks. after the .NET 8 upgrade you'll want to make sure that DB connections still work when selecting different .NET sdks from the drop-down

**Comment by tareqimbasher** [comment]
This is now complete with the merge of #352 and will go out with next NetPad release.

**Comment by tareqimbasher** [comment]
[NetPad v0.10.0](https://github.com/tareqimbasher/NetPad/releases/tag/v0.10.0) is now available!
